Generals/Code/GameEngine/Source/GameClient/GUI/ControlBar/ControlBar.cpp

```diff
@@ -2716,7 +2722,9 @@ void ControlBar::showRallyPoint( const Coord3D *loc )
 		marker->setOrientation( TheGlobalData->m_downwindAngle );//To blow down wind -- ML
 
 		// set the marker colors to that of the local player
-		Player *player = ThePlayerList->getLocalPlayer();
+		Player* player = TheControlBar->getCurrentlyViewedPlayer();
+		if (!player)
+			return;
```

Comment:
Prefer to not use returns in the middle of functions, because it makes logic eventually more difficult to follow along

Can simplify to:

```cpp
if (Player* player = TheControlBar->getCurrentlyViewedPlayer())
{
	if (TheGlobalData->m_timeOfDay == TIME_OF_DAY_NIGHT)
		marker->setIndicatorColor( player->getPlayerNightColor() );
	else
		marker->setIndicatorColor( player->getPlayerColor() );
}
```

Reply:
I typically try to draw the line at three levels of nesting. The end of the `loc == NULL` block should really be a `return` instead of an `else`. I'm also wary of putting assignments within conditions, which can be useful in specific situations (i.e. reducing loop verbosity), but is otherwise generally discouraged. I'll give the method a readability refactor.